### PR TITLE
Port dataplane driver to Golang

### PR DESCRIPTION
**Pull request by fasaxc**

### CODE REVIEW

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by fasaxc** [comment]
retest this please

**Comment by tomdee** [comment]
@fasaxc Is this PR going to simplify the dependencies that felix has on the OS? E.g. on alpine we need to install:
`ip6tables ipset iputils iproute2 conntrack-tools`

File: go/felix/felix.go [hunk 1]
```diff
@@ -170,49 +171,73 @@ configRetry:
...
-		log.WithError(err).Fatal("Failed to close parent's copy of pipe")
+	// Start up the dataplane driver.  This may be the internal go-based driver or an external
+	// one.
+	var dpDriver dataplaneDriver
+	var dpDriverCmd *exec.Cmd
+	if configParams.UseInternalDataplaneDriver {
+		log.Info("Using internal dataplane driver.")
+		markAccept := configParams.NthIPTablesMark(0)
+		markNextTier := configParams.NthIPTablesMark(1)
+		markWorkload := configParams.NthIPTablesMark(2)
```
Comment (matthewdupre) [inline go/felix/felix.go]: These would be better with some kind of next mark variable that's incremented each time.

**Comment by fasaxc** [comment]
@tomdee Sadly not, it reduces our use of those but I think they're all still needed.

**Pull request merged**